### DISCUSSION TITLE
Generate both masked fragment types and full query types.

### DISCUSSION THREAD

**Discussion by vic08** [dotansimha/graphql-code-generator#9308] [Q&A]
I'm using msw.js to mock the api responses for my tests. I'd like to use the generated types for these mocks, but currently I can't do that, because of fragment masking. Is anyone aware of a workaround to solve the issue?

**Comment by charpeni** [comment]
You'll need to unmask fragment types.

I created an issue for it here: https://github.com/dotansimha/graphql-code-generator/issues/9075. You can look at the implementation proof of concept on TypeScript Playground for `UnmaskResultOf<TypedDocumentNode>` and if you like it, I'll be happy to send a pull request for it. I thought I was the only one with that interest. 😄
